### PR TITLE
Add CI workflow

### PR DESCRIPTION
**Pull request by Leticia-maria**
TODO
- add more python versions
- run tests on repo
- should I test across multiple OS setups? @tommyh-ECL 
- run CI for pull requests

### CODE REVIEW

**Comment by Leticia-maria** [comment]
Related to #31

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```
Comment (tommyh-ECL) [inline .github/workflows/CI.yml:20]: update python version

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  pull_request:
+    branches:
+      - master
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up python 3.7
```
Comment (tommyh-ECL) [inline .github/workflows/CI.yml:17]: update python version

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Install dependencies
+      run: |
+        cd src
+        python -m pip install --upgrade pip
+        pip install coverage
+        if  [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Run Unit Tests and generate coverage report
+      run: |
+        cd src
+        python -m unittest -v
```
Comment (tommyh-ECL) [inline .github/workflows/CI.yml:30]: we use `pytest` not `unittest` for this package

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      with:
+        name: code-coverage-report
+        path: src/htmlcov
+    - name: Run package creation
+      run: |
+        cd src
+        python -m pip install --user --upgrade build
+        python -m build
+    - name: Archive package
+      uses: actions/upload-artifact@v2
```
Comment (tommyh-ECL) [inline .github/workflows/CI.yml:45]: where does this get uploaded to? will this upload the package to `pypi`? Ideally, we'd like to conditionally upload the package to pypi upon version changes in the `setup.py` file

**Review by tommyh-ECL — CHANGES_REQUESTED** [review]
minor fixes

**Comment by Leticia-maria** [comment]
I think it is ready to merge @tommyh-ECL

**Review by tommyh-ECL — APPROVED** [review]

**Pull request merged**